### PR TITLE
New test cases and few issues fix

### PR DESCRIPTION
**Pull request by akash-akya**
New test cases and few issues fix
Please review

### CODE REVIEW

File: lib/exq_scheduler/schedule/parser.ex [hunk 1]
```diff
@@ -54,6 +55,20 @@ defmodule ExqScheduler.Schedule.Parser do
...
   defp create_job(schedule) do
-    Map.drop(schedule, @non_job_keys) |> Poison.encode!()
+    job = Map.drop(schedule, @non_job_keys)
+    validate_config(job)
+    job = job |> Poison.encode!()
+  end
+
+  defp validate_config(job) do
+    if job["class"] == nil do
+      exit "class is not configured for scheduler"
```
Comment (ananthakumaran) [inline lib/exq_scheduler/schedule/parser.ex]: It's better to raise an exception with proper message. Add details like name, cron etc so it's easy to identify

File: test/exq_scheduler_test.exs [hunk 2]
```diff
@@ -19,4 +28,104 @@ defmodule ExqSchedulerTest do
...
+                                                 "cron" => "0 * * * * *",
+                                                 "class" => "QWorker",
+                                                 "queue" => "SuperQ",
+                                                 "include_metadata" => true}])
+  test "check continuity" do
+    redis = redis_pid("continuity")
+    storage_opts = Storage.build_opts(env([:redis, :name], redis))
+    :timer.sleep(4000) # 3+ Hour
+
+    jobs = get_jobs_from_storage(redis, Storage.queue_key("default", storage_opts))
```
Comment (ananthakumaran) [inline test/exq_scheduler_test.exs]: This should probably be moved to test helper, pass the queue name and class use, use the redis that's used for flushdb now.

File: test/exq_scheduler_test.exs [hunk 3]
```diff
@@ -19,4 +28,104 @@ defmodule ExqSchedulerTest do
...
+  end
+
+  defp assert_continuity(jobs, diff) do
+    schedule_times =
+      jobs
+      |> Enum.map(fn job -> List.first(job.args)["scheduled_at"] end)
+      |> Enum.map(&convert_from_iso_to_unixtime(&1))
+
+    len = length(schedule_times)
+    Enum.with_index(schedule_times)
```
Comment (ananthakumaran) [inline test/exq_scheduler_test.exs]: I guess we could use `Enum.chunk_every` here

File: test/exq_scheduler_test.exs [hunk 4]
```diff
@@ -19,4 +28,103 @@ defmodule ExqSchedulerTest do
...
+      |> Enum.filter(fn job -> job.class == "TimeWorker" end)
+
+    is_jobs_scheduled_before_start =
+      jobs
+      |> Enum.all?(
+         fn job ->
+           st = List.first(job.args)["scheduled_at"]
+           convert_from_iso_to_unixtime(st) > start_time
+         end)
+    assert(is_jobs_scheduled_before_start == true, inspect(jobs))
```
Comment (ananthakumaran) [inline test/exq_scheduler_test.exs]: == true is redundant

File: test/schedule_test.exs [hunk 5]
```diff
@@ -3,6 +3,24 @@ defmodule ScheduleTest do
   alias ExqScheduler.Time
   import TestUtils
 
+  import Logger
```
Comment (ananthakumaran) [inline test/schedule_test.exs]: is this used?

File: test/storage_test.exs [hunk 6]
```diff
@@ -27,6 +27,13 @@ defmodule StorageTest do
     assert length(schedules) == 2
   end
 
+  test "Check if default schedule state is 'ture'" do
```
Comment (ananthakumaran) [inline test/storage_test.exs]: typo

File: test/exq_scheduler_test.exs (hunk 4 above)
Comment (akash-akya) [reply test/exq_scheduler_test.exs]: You mean we can assert inside the 'Enum.all' itself?

Comment (ananthakumaran) [reply test/exq_scheduler_test.exs]: assert(x) is the same as assert(x == true)

**Pull request merged**